### DISCUSSION TITLE
Installing structure should import existing Pages

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3124] [Feature Requests]
Now that structure is native...

When you go to install Structure, if Pages is installed, you get a message to uninstall it first.  But if you have a bunch of existing pages, you don't want to lose them.  There is no way to get those pages into structure.

When you uninstall pages, the array data are unchanged.  So you haven't lost anything yet.  Go install structure, all is well.  Set up your channels and you'll find your saved Pages uri/template actually show up in the Structure tab for the entry.  They even still work frontend- you put the uri in, it still takes you to the correct template and works like it did in pages.  YAY!  

But, they don't show up in the structure tree and you get validation errors on the Structure validation pages.  BOO!

<img width="1113" alt="Screenshot 2023-03-15 at 10 09 32 AM" src="https://user-images.githubusercontent.com/1181219/225339931-5ecff1fd-7a6d-4aa0-91a6-bc5efa2726ce.png">

If you 'Use URL Routing' nothing happens.  If you 'Use Structure' it removes the old pages from the pages array and they are lost and it no longer works on the frontend.

At the least, we need to document what happens and give a better message than just saying 'go uninstall pages'.  It would be best if during structure install, it asks if you want to import the pages or delete them.  Because really, who wants validation errors long term?  Or kinda orphan pages that work, but aren't in a tree?

**Comment by dougblackjr** [comment]
+1 for this. If Structure is the standard going forward, would be excellent to automate the migration process away from Pages.

**Comment by intoeetive** [comment]
Looking into this, Structure does create records for existing pages on install, but it creates those as Listings (exp_structure_listing) not as Pages (exp_structure)
Which does not seem correct to me

**Comment by intoeetive** [comment]
See https://github.com/ExpressionEngine/ExpressionEngine/pull/3731
